pkg/sql/plan2/function/function.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2021 - 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/errno"
+	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"github.com/matrixorigin/matrixone/pkg/sql/errors"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/add"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"reflect"
+	"sync"
+)
+
+const (
+	// NoLimit is a flag that means no number limit for argument number
+	NoLimit = -1
+
+	// NullValueType is the type of const value `NULL`
+	// If input NULL as an argument, its type should be NullValue which can match each type.
+	// e.g.
+	// built_in_function(100, NULL)
+	// its argument type is [types.T_int64, NullValueType]
+	NullValueType = types.T_any
```

Comment:
You don't know any of these -- when register function, you register a built_in_function(int64, int64).    Registration does not know, or care, about 100, and null.   It cares about int64, int64.

Reply:
> 

Sorry, my bad code comments may make you confused.
what I want to say is that

null can match to every types of function parameters.

Assume that  I register a function, it has 2 overloads.
 f1(int64, int64),
 f1(int64, float32)

when I input `select f1(100, null)`, I hope it can both match 2 overloads when I search the functions.So we will export an error for this sql.

and if we have only 1 overload in our database, we will match it, and call it.